core/store/src/trie/mem/mem_trie_update.rs

```diff
@@ -160,31 +179,24 @@ impl<'a, M: ArenaMemory> GenericTrieUpdate<'a, MemTrieNodeId, FlatStateValue>
         };
 
         // Then, record disk changes if needed.
-        let Some(tracked_node_changes) = self.tracked_trie_changes.as_mut() else {
+        let Some(nodes_tracker) = self.nodes_tracker.as_mut() else {
             return flat_value;
         };
         let GenericTrieValue::MemtrieAndDisk(value) = value else {
             return flat_value;
         };
-        tracked_node_changes
-            .refcount_inserted_values
-            .entry(value)
-            .and_modify(|rc| *rc += 1)
-            .or_insert(1);
+        nodes_tracker.refcount_inserted_values.entry(value).and_modify(|rc| *rc += 1).or_insert(1);
```

Comment:
Quick question, isn't it incorrect to return early for the `GenericTrieValue::MemtrieAndDisk` case above? Why would we not want to track those nodes?

Reply:
Oh yeah, that's confusing method.
In fact, we return early in MemtrieOnly node, not MemtrieAndDisk. We do it because we have don't have full value, so we can't compute its hash and we can't make a record.
Ideally GenericTrieValue should be consistent with `nodes_tracker` by design. That's a TODO for the future.

Made a little reordering to check `GenericTrieValue` enum variant only once.